utils/measure/measure/util/measure_util.py

```diff
@@ -21,21 +27,115 @@ def __init__(self, power_meter: PowerMeter, config: MeasureConfig) -> None:
+        resistive value is set, the power consumption of the dummy load calculated for each
+        measurement, depending on the current voltage and is subtracted from the power
+        measurements.
+
+        Args:
+            duration (int): The time duration (in seconds) over which to take measurements.
+            measure_resistance (bool): Whether to measure resistance instead of power. Defaults to False.
+
+        Returns:
+            float: The average resistance (in Ω) or power (in W) over the measurement duration.
+
+        Raises:
+            UnsupportedFeatureError: If `measure_resistance` is True but the power meter does not
+                                      support voltage measurements.
+
+        Error handling:
+            - If voltage measurements are not supported, the program will terminate with an appropriate
+              error message, if measuring power consumption and a dummy load resistive value is set.
+            - For resistance measurements, voltage values below 1 are treated as invalid, and the program
+              exits to avoid incorrect calculations.
+            - Ignores single measurements of <= 0 W.
+        """
+        _LOGGER.info("Measuring average %s over %s seconds", "resistance" if measure_resistance else "power", duration)
         start_time = time.time()
         readings: list[float] = []
+
+        first_measurement = True
+
         while (time.time() - start_time) < duration:
+            if first_measurement:
+                first_measurement = False
+            else:
+                # sleep time exceeds duration
+                if not ((time.time() - start_time + self.config.sleep_time) < duration):
+                    break
+                time.sleep(self.config.sleep_time)
+
+            if measure_resistance:
+                result = self.power_meter.get_power(include_voltage=True)
+                power, voltage = result.power, result.voltage
+
+                if voltage < 1:
+                    _LOGGER.error("Error during measurement: Voltage measurement returned zero. Aborting measurement.")
+                    exit(1)
+
+                if round(power, 2) == 0:
+                    _LOGGER.warning("Invalid measurement: power: %.2f W, voltage: %.2f", power, voltage)
+                    continue
+
+                resistance = round((voltage**2) / power, 4)
+                readings.append(resistance)
+                _LOGGER.info(
+                    "Measured resistance: %.2f Ω; measured power: %.2f W, voltage: %.2f",
+                    resistance,
+                    power,
+                    voltage,
+                )
+                continue
+
+            if self.dummy_load_value:  # measurement with dummy load
+                try:
+                    result = self.power_meter.get_power(include_voltage=True)
+                except UnsupportedFeatureError as e:
```

Comment:
Currenty power meter support is checked here but also in trend calculation logic.
Cleaner to check voltage support only one time when dummy load has been selected, and abort early there.
Might add method `is_voltage_measurement_supported` to the powermeter abstract base class, that would also cleanup the measure logic in the powermeters a bit, so we don't need if/else statements there.

Reply:
I've added this via 9b57cbe54c1349617f860d201798c9e0ce010d3a, but I don't see the if/else cleanup potential. Maybe I'm just too tired.